neuromation/cli/storage.py

```diff
@@ -835,6 +838,67 @@ def filter_option(*args: str, flag_value: bool, help: str) -> Callable[[Any], An
         sys.exit(EX_OSFILE)
 
 
+@command()
+@click.argument("path", required=False)
+@option(
+    "-a",
+    "--all",
+    "show_all",
+    is_flag=True,
+    help="do not ignore entries starting with .",
+)
+@option(
+    "--human-readable",
+    "-h",
+    is_flag=True,
+    help="Print the size in a more human readable way.",
+)
+@option(
+    "--size", "-s", is_flag=True, help="Print the size in bytes of each file.",
+)
+@option(
+    "--sort",
+    type=click.Choice(["name", "size", "time"]),
+    default="name",
+    help="sort by given field, default is name",
+)
+async def tree(
+    root: Root, path: str, size: bool, human_readable: bool, sort: str, show_all: bool
+) -> None:
+    """List contents of directories in a tree-like format.
+
+    Tree is a recursive directory listing program that produces a depth indented listing
+    of files, which is colorized ala dircolors if the LS_COLORS environment variable is
+    set and output is to tty.  With no arguments, tree lists the files in the storage:
+    directory.  When directory arguments are given, tree lists all the files and/or
+    directories found in the given directories each in turn.  Upon completion of listing
+    all files/directories found, tree returns the total number of files and/or
+    directories listed.
+
+    By default PATH is equal user's home dir (storage:)
+
+    """
+    if not path:
+        path = "storage:"
+    uri = parse_file_resource(path, root)
+
+    errors = False
+    try:
+        tree = await fetch_tree(root.client, uri, show_all)
+        tree = dataclasses.replace(tree, name=str(path))
+    except (OSError, ResourceNotFound, IllegalArgumentError) as error:
```

Comment:
I am not sure that `IllegalArgumentError` is still needed. It was raised when you tried to get a listdir of a non-directory. Now it should raise `OSError` or `ResourceNotFound`.

Reply:
Ok, removed.